Module4

harbymoo opened the pull request:
Course work

Commit:
update new files

harbymoo committed:
Merge pull request #6 from harbymoo/main

update module 4 latest changes

Commit:
updated

Commit:
filename change
updated ansible.cfg to include cfg file

Commit:
updated playbook.

Commit:
tagged the debug tasks
updated README

Commit:
Merge remote-tracking branch 'origin/module4' into module4
pushing to remote branch prompted this

Commit:
updated

JackMeadDev commented on Module1/ansible/hosts:
```diff
@@ -0,0 +1,2 @@
+[module4]
+mod4node ansible_host=35.179.70.200
```
I'm not able to see the app running on this node, but I can see from the service logs that it was previously - so I'm assuming you went ahead and turned it off!

JackMeadDev commented on Module1/ansible/teardown_playbook.yml:
```diff
@@ -0,0 +1,98 @@
+---
```
Ah, the existence of this file answers my question! Nice addition 😄

JackMeadDev reviewed (approved):
Nice work, I can see that your app was successfully deployed, and you've gone well beyond the requirements with all the stretch tasks (including the clean up one - nice addition!)

Very happy for you to merge this when you're ready

By default I'll schedule for these VMs to be torn down at the end of next week as we won't be using them again going forward

Edit: And with the password you provided on Aptem I was able to run it successfully too - thanks!

JackMeadDev commented on Module1/ansible/module4_playbook.yml:
```diff
@@ -0,0 +1,178 @@
+        {{ POETRY_BIN }} install --directory=/opt/todoapp/Module1
+      register: poetry_update_output
+      become: yes
+      become_user: "{{ POETRY_USER }}"
```
Interestingly this step errors for me if I run it a second time with "Permission Denied"
> [Permission denied](fatal: [mod4node]: FAILED! => {"changed": true, "cmd": "/home/todoapp/.local/bin/poetry install --directory=/opt/todoapp/Module1\n", "delta": "0:00:01.348117", "end": "2023-11-14 09:51:30.551134", "msg": "non-zero return code", "rc": 1, "start": "2023-11-14 09:51:29.203017", "stderr": "\n[Errno 13] Permission denied: '/opt/todoapp/Module1/.venv/lib/python3.9/site-packages/todo_app.pth'", "stderr_lines": ["", "[Errno 13] Permission denied: '/opt/todoapp/Module1/.venv/lib/python3.9/site-packages/todo_app.pth'"], "stdout": "Installing dependencies from lock file\n\nNo dependencies to install or update\n\nInstalling the current project: todo-app (0.1.0)", "stdout_lines": ["Installing dependencies from lock file", "", "No dependencies to install or update", "", "Installing the current project: todo-app (0.1.0)"]})

There might be an issue with this not being idempotent somewhere

harbymoo commented:
Please try again. Seems that I was logged in and in the /opt/todoapp/.

On Tue, Nov 14, 2023 at 9:53 AM JackMead ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Module1/ansible/module4_playbook.yml
> <https://github.com/harbymoo/DevOps-Course-Starter/pull/7#discussion_r1392306008>
> :
>
> > +        update: no
> +      register: git_output
> +      notify: update permissions
> +
> +    - name: output git clone
> +      debug:
> +        msg: "{{ git_output }}"
> +      tags: debug_output
> +
> +    # update the poetry
> +    - name: poetry update
> +      shell: |
> +        {{ POETRY_BIN }} install --directory=/opt/todoapp/Module1
> +      register: poetry_update_output
> +      become: yes
> +      become_user: "{{ POETRY_USER }}"
>
> Interestingly this step errors for me if I run it a second time with
> "Permission Denied"
>
> [Permission denied](fatal: [mod4node]: FAILED! => {"changed": true, "cmd":
> "/home/todoapp/.local/bin/poetry install
> --directory=/opt/todoapp/Module1\n", "delta": "0:00:01.348117", "end":
> "2023-11-14 09:51:30.551134", "msg": "non-zero return code", "rc": 1,
> "start": "2023-11-14 09:51:29.203017", "stderr": "\n[Errno 13] Permission
> denied:
> '/opt/todoapp/Module1/.venv/lib/python3.9/site-packages/todo_app.pth'",
> "stderr_lines": ["", "[Errno 13] Permission denied:
> '/opt/todoapp/Module1/.venv/lib/python3.9/site-packages/todo_app.pth'"],
> "stdout": "Installing dependencies from lock file\n\nNo dependencies to
> install or update\n\nInstalling the current project: todo-app (0.1.0)",
> "stdout_lines": ["Installing dependencies from lock file", "", "No
> dependencies to install or update", "", "Installing the current project:
> todo-app (0.1.0)"]})
>
> There might be an issue with this not being idempotent somewhere
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/harbymoo/DevOps-Course-Starter/pull/7#pullrequestreview-1729340003>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKGN7O6WB3Y65VHXOKMD653YEM5QFAVCNFSM6AAAAAA7JBF5IOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMRZGM2DAMBQGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

JackMeadDev commented:
Tried again and I see the same issue, I think it's related to:
* When GitHub clones the files, the permissions are set to one thing
* Then we run the initial `poetry install` as the poetry user
* _Then_ the handler gets run afterwards that updates the permissions

So next time the `poetry install` is run with the permissions set in the handler, rather than in the initial clone
I suspect the simplest solution is making sure the "group" has write permissions in the permissions handler, i.e. `775` rather than `755` - that seemed to work on a test
Because of that, I suspect a longer term/better solution would be to have the permissions be an explicit task rather than a handler, so that we're confident that it's always run before the `poetry install` command, and that executes on the same permissions each time

harbymoo commented:
This works when run as the user, ec2-user, without issue.

***@***.*** ~]$ sudo lsof -i:5000
COMMAND     PID    USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
flask   1275056 todoapp    3u  IPv4 6789418      0t0  TCP *:commplex-main
(LISTEN)
python  1275169 todoapp    3u  IPv4 6789418      0t0  TCP *:commplex-main
(LISTEN)
python  1275169 todoapp    4u  IPv4 6789418      0t0  TCP *:commplex-main
(LISTEN)

***@***.*** ~]$ curl -I localhost:5000
HTTP/1.1 200 OK
Server: Werkzeug/2.3.7 Python/3.9.16
Date: Tue, 14 Nov 2023 18:35:19 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 7138
Connection: close

On Tue, Nov 14, 2023 at 10:57 AM JackMead ***@***.***> wrote:

> Tried again and I see the same issue, I think it's related to:
>
>    - When GitHub clones the files, the permissions are set to one thing
>    - Then we run the initial poetry install as the poetry user
>    - *Then* the handler gets run afterwards that updates the permissions
>
> So next time the poetry install is run with the permissions set in the
> handler, rather than in the initial clone
> I suspect the simplest solution is making sure the "group" has write
> permissions in the permissions handler, i.e. 775 rather than 755 - that
> seemed to work on a test
> Because of that, I suspect a longer term/better solution would be to have
> the permissions be an explicit task rather than a handler, so that we're
> confident that it's always running the poetry install on the same
> permissions
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/harbymoo/DevOps-Course-Starter/pull/7#issuecomment-1809985289>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKGN7O7FJJIRTAWHGBRRVZDYENFDBAVCNFSM6AAAAAA7JBF5IOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMBZHE4DKMRYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


PLAY [module4] *****************************************************************

TASK [create user todoapp] *****************************************************
changed: [mod4node]

TASK [debug for user] **********************************************************
ok: [mod4node] => {
    "useradd_output": {
        "ansible_facts": {
            "discovered_interpreter_python": "/usr/bin/python3.9"
        },
        "changed": true,
        "comment": "APP user",
        "create_home": true,
        "failed": false,
        "group": 1000,
        "home": "/home/todoapp",
        "name": "todoapp",
        "shell": "/bin/bash",
        "state": "present",
        "system": false,
        "uid": 1001,
        "warnings": [
            "Platform linux on host mod4node is using the discovered Python interpreter at /usr/bin/python3.9, but future installation of another Python interpreter could change the meaning of that path. See https://docs.ansible.com/ansible-core/2.15/reference_appendices/interpreter_discovery.html for more information."
        ]
    }
}

TASK [package installation] ****************************************************
changed: [mod4node]

TASK [debug] *******************************************************************
ok: [mod4node] => {
    "msg": {
        "ansible_facts": {
            "pkg_mgr": "dnf"
        },
        "changed": true,
        "failed": false,
        "msg": "",
        "rc": 0,
        "results": [
            "Installed: git-2.40.1-1.amzn2023.0.1.x86_64",
            "Installed: python3.11-3.11.2-2.amzn2023.0.10.x86_64",
            "Installed: perl-Git-2.40.1-1.amzn2023.0.1.noarch"
        ]
    }
}

TASK [installation of poetry] **************************************************
changed: [mod4node]

TASK [debug] *******************************************************************
ok: [mod4node] => {
    "msg": {
        "changed": true,
        "cmd": "curl -sSL https://install.python-poetry.org | python3 -\n",
        "delta": "0:00:15.692794",
        "end": "2023-11-14 18:27:52.560831",
        "failed": false,
        "msg": "",
        "rc": 0,
        "start": "2023-11-14 18:27:36.868037",
        "stderr": "",
        "stderr_lines": [],
        "stdout": "Retrieving Poetry metadata\n\n# Welcome to Poetry!\n\nThis will download and install the latest version of Poetry,\na dependency and package manager for Python.\n\nIt will add the `poetry` command to Poetry's bin directory, located at:\n\n/home/todoapp/.local/bin\n\nYou can uninstall at any time by executing this script with the --uninstall option,\nand these changes will be reverted.\n\nInstalling Poetry (1.7.0)\nInstalling Poetry (1.7.0): Creating environment\nInstalling Poetry (1.7.0): Installing Poetry\nInstalling Poetry (1.7.0): Creating script\nInstalling Poetry (1.7.0): Done\n\nPoetry (1.7.0) is installed now. Great!\n\nTo get started you need Poetry's bin directory (/home/todoapp/.local/bin) in your `PATH`\nenvironment variable.\n\nAdd `export PATH=\"/home/todoapp/.local/bin:$PATH\"` to your shell configuration file.\n\nAlternatively, you can call Poetry explicitly with `/home/todoapp/.local/bin/poetry`.\n\nYou can test that everything is set up by executing:\n\n`poetry --version`",
        "stdout_lines": [
            "Retrieving Poetry metadata",
            "",
            "# Welcome to Poetry!",
            "",
            "This will download and install the latest version of Poetry,",
            "a dependency and package manager for Python.",
            "",
            "It will add the `poetry` command to Poetry's bin directory, located at:",
            "",
            "/home/todoapp/.local/bin",
            "",
            "You can uninstall at any time by executing this script with the --uninstall option,",
            "and these changes will be reverted.",
            "",
            "Installing Poetry (1.7.0)",
            "Installing Poetry (1.7.0): Creating environment",
            "Installing Poetry (1.7.0): Installing Poetry",
            "Installing Poetry (1.7.0): Creating script",
            "Installing Poetry (1.7.0): Done",
            "",
            "Poetry (1.7.0) is installed now. Great!",
            "",
            "To get started you need Poetry's bin directory (/home/todoapp/.local/bin) in your `PATH`",
            "environment variable.",
            "",
            "Add `export PATH=\"/home/todoapp/.local/bin:$PATH\"` to your shell configuration file.",
            "",
            "Alternatively, you can call Poetry explicitly with `/home/todoapp/.local/bin/poetry`.",
            "",
            "You can test that everything is set up by executing:",
            "",
            "`poetry --version`"
        ],
        "warnings": [
            "Module remote_tmp /home/todoapp/.ansible/tmp did not exist and was created with a mode of 0700, this may cause issues when running as another user. To avoid this, create the remote_tmp dir with the correct permissions manually"
        ]
    }
}

TASK [create directory] ********************************************************
changed: [mod4node]

TASK [dir output] **************************************************************
ok: [mod4node] => {
    "msg": {
        "changed": true,
        "diff": {
            "after": {
                "group": 1000,
                "owner": 1000,
                "path": "/opt/todoapp",
                "state": "directory"
            },
            "before": {
                "group": 0,
                "owner": 0,
                "path": "/opt/todoapp",
                "state": "absent"
            }
        },
        "failed": false,
        "gid": 1000,
        "group": "ec2-user",
        "mode": "0755",
        "owner": "ec2-user",
        "path": "/opt/todoapp",
        "secontext": "unconfined_u:object_r:usr_t:s0",
        "size": 6,
        "state": "directory",
        "uid": 1000
    }
}

TASK [Example clone of a single branch] ****************************************
changed: [mod4node]

TASK [output git clone] ********************************************************
ok: [mod4node] => {
    "msg": {
        "after": "07a51775e790b0c8ed5b7494ea15b0d4bc1d0270",
        "before": null,
        "changed": true,
        "failed": false
    }
}

TASK [poetry update] ***********************************************************
changed: [mod4node]

TASK [debug] *******************************************************************
ok: [mod4node] => {
    "msg": {
        "changed": true,
        "cmd": "/home/todoapp/.local/bin/poetry install --directory=/opt/todoapp/Module1\n",
        "delta": "0:00:02.599909",
        "end": "2023-11-14 18:27:59.868551",
        "failed": false,
        "msg": "",
        "rc": 0,
        "start": "2023-11-14 18:27:57.268642",
        "stderr": "Creating virtualenv todo-app in /opt/todoapp/Module1/.venv",
        "stderr_lines": [
            "Creating virtualenv todo-app in /opt/todoapp/Module1/.venv"
        ],
        "stdout": "Installing dependencies from lock file\n\nPackage operations: 21 installs, 0 updates, 0 removals\n\n  • Installing markupsafe (2.1.3)\n  • Installing zipp (3.17.0)\n  • Installing blinker (1.6.2)\n  • Installing certifi (2023.7.22)\n  • Installing charset-normalizer (3.2.0)\n  • Installing click (8.1.7)\n  • Installing exceptiongroup (1.1.3)\n  • Installing idna (3.4)\n  • Installing importlib-metadata (6.8.0)\n  • Installing iniconfig (2.0.0)\n  • Installing itsdangerous (2.1.2)\n  • Installing jinja2 (3.1.2)\n  • Installing packaging (23.1)\n  • Installing pluggy (1.3.0)\n  • Installing tomli (2.0.1)\n  • Installing urllib3 (2.0.5)\n  • Installing werkzeug (2.3.7)\n  • Installing flask (2.3.3)\n  • Installing pytest (7.4.2)\n  • Installing python-dotenv (0.14.0)\n  • Installing requests (2.31.0)\n\nInstalling the current project: todo-app (0.1.0)",
        "stdout_lines": [
            "Installing dependencies from lock file",
            "",
            "Package operations: 21 installs, 0 updates, 0 removals",
            "",
            "  • Installing markupsafe (2.1.3)",
            "  • Installing zipp (3.17.0)",
            "  • Installing blinker (1.6.2)",
            "  • Installing certifi (2023.7.22)",
            "  • Installing charset-normalizer (3.2.0)",
            "  • Installing click (8.1.7)",
            "  • Installing exceptiongroup (1.1.3)",
            "  • Installing idna (3.4)",
            "  • Installing importlib-metadata (6.8.0)",
            "  • Installing iniconfig (2.0.0)",
            "  • Installing itsdangerous (2.1.2)",
            "  • Installing jinja2 (3.1.2)",
            "  • Installing packaging (23.1)",
            "  • Installing pluggy (1.3.0)",
            "  • Installing tomli (2.0.1)",
            "  • Installing urllib3 (2.0.5)",
            "  • Installing werkzeug (2.3.7)",
            "  • Installing flask (2.3.3)",
            "  • Installing pytest (7.4.2)",
            "  • Installing python-dotenv (0.14.0)",
            "  • Installing requests (2.31.0)",
            "",
            "Installing the current project: todo-app (0.1.0)"
        ]
    }
}

TASK [Create .env from template] ***********************************************
changed: [mod4node]

TASK [debug template] **********************************************************
ok: [mod4node] => {
    "msg": {
        "changed": true,
        "checksum": "a5f2e1fc1a1a52cd9377b05d0d0854c8a561af09",
        "dest": "/opt/todoapp/Module1/.env",
        "diff": [],
        "failed": false,
        "gid": 1000,
        "group": "ec2-user",
        "md5sum": "875dbd4910be113b2c6741a4f996d1cc",
        "mode": "0664",
        "owner": "ec2-user",
        "secontext": "system_u:object_r:usr_t:s0",
        "size": 759,
        "src": "/home/ec2-user/.ansible/tmp/ansible-tmp-1699986480.209277-1234423-78975582999234/source",
        "state": "file",
        "uid": 1000
    }
}

TASK [copy the service file] ***************************************************
changed: [mod4node]

TASK [debug service] ***********************************************************
ok: [mod4node] => {
    "service_copy_output": {
        "changed": true,
        "checksum": "9f67a8933633da0667f8a89189ac481fdde766e4",
        "dest": "/etc/systemd/system/todoapp.service",
        "diff": [],
        "failed": false,
        "gid": 0,
        "group": "root",
        "md5sum": "4c20d64a87744c4c88e052e1e0351e84",
        "mode": "0644",
        "owner": "root",
        "secontext": "system_u:object_r:systemd_unit_file_t:s0",
        "size": 222,
        "src": "/home/ec2-user/.ansible/tmp/ansible-tmp-1699986482.1203408-1234441-69575290156125/source",
        "state": "file",
        "uid": 0
    }
}

TASK [systemd enable] **********************************************************
changed: [mod4node]

TASK [debug systemd] ***********************************************************
ok: [mod4node] => {
    "todoapp_systemd_output": {
        "changed": true,
        "enabled": true,
        "failed": false,
        "name": "todoapp.service",
        "state": "started",
        "status": {
            "AccessSELinuxContext": "system_u:object_r:systemd_unit_file_t:s0",
            "ActiveEnterTimestampMonotonic": "0",
            "ActiveExitTimestampMonotonic": "0",
            "ActiveState": "inactive",
            "After": "basic.target systemd-journald.socket -.mount system.slice sysinit.target",
            "AllowIsolate": "no",
            "AssertResult": "no",
            "AssertTimestampMonotonic": "0",
            "Before": "shutdown.target",
            "BlockIOAccounting": "no",
            "BlockIOWeight": "[not set]",
            "CPUAccounting": "yes",
            "CPUAffinityFromNUMA": "no",
            "CPUQuotaPerSecUSec": "infinity",
            "CPUQuotaPeriodUSec": "infinity",
            "CPUSchedulingPolicy": "0",
            "CPUSchedulingPriority": "0",
            "CPUSchedulingResetOnFork": "no",
            "CPUShares": "[not set]",
            "CPUUsageNSec": "[not set]",
            "CPUWeight": "[not set]",
            "CacheDirectoryMode": "0755",
            "CanFreeze": "yes",
            "CanIsolate": "no",
            "CanReload": "no",
            "CanStart": "yes",
            "CanStop": "yes",
            "CapabilityBoundingSet": "cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setuid cap_setpcap cap_linux_immutable cap_net_bind_service cap_net_broadcast cap_net_admin cap_net_raw cap_ipc_lock cap_ipc_owner cap_sys_module cap_sys_rawio cap_sys_chroot cap_sys_ptrace cap_sys_pacct cap_sys_admin cap_sys_boot cap_sys_nice cap_sys_resource cap_sys_time cap_sys_tty_config cap_mknod cap_lease cap_audit_write cap_audit_control cap_setfcap cap_mac_override cap_mac_admin cap_syslog cap_wake_alarm cap_block_suspend cap_audit_read cap_perfmon cap_bpf cap_checkpoint_restore",
            "CleanResult": "success",
            "CollectMode": "inactive",
            "ConditionResult": "no",
            "ConditionTimestampMonotonic": "0",
            "ConfigurationDirectoryMode": "0755",
            "Conflicts": "shutdown.target",
            "ControlGroupId": "0",
            "ControlPID": "0",
            "CoredumpFilter": "0x33",
            "DefaultDependencies": "yes",
            "DefaultMemoryLow": "0",
            "DefaultMemoryMin": "0",
            "Delegate": "no",
            "Description": "todoapp.service",
            "DevicePolicy": "auto",
            "DropInPaths": "/etc/systemd/system/todoapp.service.d/override.conf",
            "DynamicUser": "no",
            "ExecMainCode": "0",
            "ExecMainExitTimestampMonotonic": "0",
            "ExecMainPID": "0",
            "ExecMainStartTimestampMonotonic": "0",
            "ExecMainStatus": "0",
            "ExecStart": "{ path=/home/todoapp/.local/bin/poetry ; argv[]=/home/todoapp/.local/bin/poetry run flask run --host=0.0.0.0 ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }",
            "ExecStartEx": "{ path=/home/todoapp/.local/bin/poetry ; argv[]=/home/todoapp/.local/bin/poetry run flask run --host=0.0.0.0 ; flags= ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }",
            "ExitType": "main",
            "FailureAction": "none",
            "FileDescriptorStoreMax": "0",
            "FinalKillSignal": "9",
            "FragmentPath": "/etc/systemd/system/todoapp.service",
            "FreezerState": "running",
            "GID": "[not set]",
            "GuessMainPID": "yes",
            "IOAccounting": "no",
            "IOReadBytes": "18446744073709551615",
            "IOReadOperations": "18446744073709551615",
            "IOSchedulingClass": "2",
            "IOSchedulingPriority": "4",
            "IOWeight": "[not set]",
            "IOWriteBytes": "18446744073709551615",
            "IOWriteOperations": "18446744073709551615",
            "IPAccounting": "no",
            "IPEgressBytes": "[no data]",
            "IPEgressPackets": "[no data]",
            "IPIngressBytes": "[no data]",
            "IPIngressPackets": "[no data]",
            "Id": "todoapp.service",
            "IgnoreOnIsolate": "no",
            "IgnoreSIGPIPE": "yes",
            "InactiveEnterTimestampMonotonic": "0",
            "InactiveExitTimestampMonotonic": "0",
            "JobRunningTimeoutUSec": "infinity",
            "JobTimeoutAction": "none",
            "JobTimeoutUSec": "infinity",
            "KeyringMode": "private",
            "KillMode": "control-group",
            "KillSignal": "15",
            "LimitAS": "infinity",
            "LimitASSoft": "infinity",
            "LimitCORE": "infinity",
            "LimitCORESoft": "infinity",
            "LimitCPU": "infinity",
            "LimitCPUSoft": "infinity",
            "LimitDATA": "infinity",
            "LimitDATASoft": "infinity",
            "LimitFSIZE": "infinity",
            "LimitFSIZESoft": "infinity",
            "LimitLOCKS": "infinity",
            "LimitLOCKSSoft": "infinity",
            "LimitMEMLOCK": "infinity",
            "LimitMEMLOCKSoft": "infinity",
            "LimitMSGQUEUE": "819200",
            "LimitMSGQUEUESoft": "819200",
            "LimitNICE": "0",
            "LimitNICESoft": "0",
            "LimitNOFILE": "65535",
            "LimitNOFILESoft": "65535",
            "LimitNPROC": "infinity",
            "LimitNPROCSoft": "infinity",
            "LimitRSS": "infinity",
            "LimitRSSSoft": "infinity",
            "LimitRTPRIO": "0",
            "LimitRTPRIOSoft": "0",
            "LimitRTTIME": "infinity",
            "LimitRTTIMESoft": "infinity",
            "LimitSIGPENDING": "30446",
            "LimitSIGPENDINGSoft": "30446",
            "LimitSTACK": "10485760",
            "LimitSTACKSoft": "10485760",
            "LoadState": "loaded",
            "LockPersonality": "no",
            "LogLevelMax": "-1",
            "LogRateLimitBurst": "0",
            "LogRateLimitIntervalUSec": "0",
            "LogsDirectoryMode": "0755",
            "MainPID": "0",
            "ManagedOOMMemoryPressure": "auto",
            "ManagedOOMMemoryPressureLimit": "0",
            "ManagedOOMPreference": "none",
            "ManagedOOMSwap": "auto",
            "MemoryAccounting": "yes",
            "MemoryAvailable": "infinity",
            "MemoryCurrent": "[not set]",
            "MemoryDenyWriteExecute": "no",
            "MemoryHigh": "infinity",
            "MemoryLimit": "infinity",
            "MemoryLow": "0",
            "MemoryMax": "infinity",
            "MemoryMin": "0",
            "MemorySwapMax": "infinity",
            "MountAPIVFS": "no",
            "NFileDescriptorStore": "0",
            "NRestarts": "0",
            "NUMAPolicy": "n/a",
            "Names": "todoapp.service",
            "NeedDaemonReload": "no",
            "Nice": "0",
            "NoNewPrivileges": "no",
            "NonBlocking": "no",
            "NotifyAccess": "none",
            "OOMPolicy": "stop",
            "OOMScoreAdjust": "0",
            "OnFailureJobMode": "replace",
            "OnSuccessJobMode": "fail",
            "Perpetual": "no",
            "PrivateDevices": "no",
            "PrivateIPC": "no",
            "PrivateMounts": "no",
            "PrivateNetwork": "no",
            "PrivateTmp": "no",
            "PrivateUsers": "no",
            "ProcSubset": "all",
            "ProtectClock": "no",
            "ProtectControlGroups": "no",
            "ProtectHome": "no",
            "ProtectHostname": "no",
            "ProtectKernelLogs": "no",
            "ProtectKernelModules": "no",
            "ProtectKernelTunables": "no",
            "ProtectProc": "default",
            "ProtectSystem": "no",
            "RefuseManualStart": "no",
            "RefuseManualStop": "no",
            "ReloadResult": "success",
            "RemainAfterExit": "no",
            "RemoveIPC": "no",
            "Requires": "-.mount system.slice sysinit.target",
            "RequiresMountsFor": "/opt/todoapp/Module1",
            "Restart": "no",
            "RestartKillSignal": "15",
            "RestartUSec": "100ms",
            "RestrictNamespaces": "no",
            "RestrictRealtime": "no",
            "RestrictSUIDSGID": "no",
            "Result": "success",
            "RootDirectoryStartOnly": "no",
            "RuntimeDirectoryMode": "0755",
            "RuntimeDirectoryPreserve": "no",
            "RuntimeMaxUSec": "infinity",
            "RuntimeRandomizedExtraUSec": "0",
            "SameProcessGroup": "no",
            "SecureBits": "0",
            "SendSIGHUP": "no",
            "SendSIGKILL": "yes",
            "Slice": "system.slice",
            "StandardError": "inherit",
            "StandardInput": "null",
            "StandardOutput": "journal",
            "StartLimitAction": "none",
            "StartLimitBurst": "5",
            "StartLimitIntervalUSec": "10s",
            "StartupBlockIOWeight": "[not set]",
            "StartupCPUShares": "[not set]",
            "StartupCPUWeight": "[not set]",
            "StartupIOWeight": "[not set]",
            "StateChangeTimestampMonotonic": "0",
            "StateDirectoryMode": "0755",
            "StatusErrno": "0",
            "StopWhenUnneeded": "no",
            "SubState": "dead",
            "SuccessAction": "none",
            "SyslogFacility": "3",
            "SyslogLevel": "6",
            "SyslogLevelPrefix": "yes",
            "SyslogPriority": "30",
            "SystemCallErrorNumber": "2147483646",
            "TTYReset": "no",
            "TTYVHangup": "no",
            "TTYVTDisallocate": "no",
            "TasksAccounting": "yes",
            "TasksCurrent": "[not set]",
            "TasksMax": "464",
            "TimeoutAbortUSec": "1min 30s",
            "TimeoutCleanUSec": "infinity",
            "TimeoutStartFailureMode": "terminate",
            "TimeoutStartUSec": "1min 30s",
            "TimeoutStopFailureMode": "terminate",
            "TimeoutStopUSec": "1min 30s",
            "TimerSlackNSec": "50000",
            "Transient": "no",
            "Type": "simple",
            "UID": "[not set]",
            "UMask": "0022",
            "UnitFilePreset": "disabled",
            "UnitFileState": "static",
            "User": "todoapp",
            "UtmpMode": "init",
            "WatchdogSignal": "6",
            "WatchdogTimestampMonotonic": "0",
            "WatchdogUSec": "infinity",
            "WorkingDirectory": "/opt/todoapp/Module1"
        }
    }
}

RUNNING HANDLER [restart todoapp] **********************************************
changed: [mod4node]

RUNNING HANDLER [update permissions] *******************************************
changed: [mod4node]

PLAY RECAP *********************************************************************
mod4node                   : ok=20   changed=11   unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

JackMeadDev commented:
Switching to the ec2-user also sounds reasonable yep 🙂
